service-workers/service-worker/resources/request-end-to-end-worker.js

```diff
@@ -4,6 +4,8 @@ onmessage = function(e) {
   var message = e.data;
   if (typeof message === 'object' && 'port' in message) {
     port = message.port;
+
+    port.postMessage('received port');
   }
 };
```

Comment:
I've pushed up a commit to address this. I'm using a custom "pending thenable" for this purpose in order to limit the number of features in this test. If you'd prefer to use a true Promise instance, say the word and I'll be happy to do so.